docs/proposals/dynamic-kubelet-settings.md

```diff
@@ -0,0 +1,129 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for improving Kubelet config by versioning it and making it manageable via the API server.
+
+## Motivation
+
+The Kubelet is currently configured via command-line flags. This is painful for a number of reasons:
+- It makes it difficult to change the way Kubelets are configured in a running cluster, because it is often tedious to change the Kubelet startup configuration.
+- It makes it difficult to manage different Kubelet configurations for different nodes, e.g. if you want to canary a new config.
+- The current lack of versioned Kubelet configuration means that any time we change Kubelet flags, we risk breaking someone's setup.
+- Having the ability to pass config in the same format on-disk as via the API server will make it easier to bootstrap master nodes before the API server is available.
+
+
+## Goals of this design:
+
+- Add a well-defined, versioned configmap object for Kubelet configuration.
+- Add/Remove/Update global and per-node Kubelet config on the fly via the API server.
+- Provide a reliable way to get the configuration of a given Kubelet along with its source (remote or on disk or flags).
+- We want the Kubelet to work with the built-in defaults on all distros we test against.
```

Comment:
As far as I remember, the defaults applied in `pkg/apis/componentconfig/v1alpha1/defaults.go` were in line with the values in `cmd/kubelet/app/options/options.go` when #19837 went in. Flags were, and still are, applied after these values, so no sets of flags should be overridden by the changes proposed here, if that's what you mean.